napi_js_data.md

```diff
@@ -0,0 +1,760 @@
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript Number
+- `[out] result`: C++ int32 primitive equivalent of the given JavaScript Number
+
+#### Return value
+- `napi_ok` if the API succeeded.
+- `napi_number_expected` if a non-number `napi_value` is passed in
+
+### *napi_get_value_int64*
+
+#### Signature
+```
+napi_status napi_get_value_int64(napi_env env, napi_value value, int64_t* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript Number
+- `[out] result`: C++ int64 primitive equivalent of the given JavaScript Number
+
+#### Return value
+- `napi_ok` if the API succeeded.
+- `napi_number_expected` if a non-number `napi_value` is passed in
+
+### *napi_get_value_string_length*
+
+#### Signature
+```
+napi_status napi_get_value_string_length(napi_env env, napi_value value, int* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript string
+- `[out] result`: Number of characters in the given JavaScript string
+
+#### Return value
+- `napi_ok` if the API succeeded.
+- `napi_string_expected` if a non-string `napi_value` is passed in
+
+### *napi_get_value_string_utf8*
+
+#### Signature
+```
+napi_status napi_get_value_string_utf8(napi_env env,
+                                       napi_value value,
+                                       char* buf,
+                                       size_t bufsize,
+                                       size_t* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript string
+- `[in]  buf`: Buffer to write the UTF8 encoded string into. If NULL is passed in, the length of the string (in bytes) is returned
+- `[in]  bufsize`: Size of the destination buffer. 
+- `[out] result`: Number of bytes copied into the buffer including the null terminateor. 
+If the buffer size is insufficient, the string will be truncated including a null terminator.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+- `napi_string_expected` if a non-string `napi_value` is passed in
+
+### *napi_get_value_string_utf16_length*
+
+#### Signature
+```
+napi_status napi_get_value_string_utf16(napi_env env,
+                                        napi_value value,
+                                        char16_t* buf,
+                                        size_t bufsize,
+                                        size_t* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: `napi_value` representing JavaScript string
+- `[in]  buf`: Buffer to write the UTF16-LE encoded string into. If NULL is passed in, the length of the string (in bytes) is returned
```

Comment:
If null is passed in, this returns the length of the string _in 2-byte code units_.